dd-sdk-android-ndk/src/androidTest/kotlin/com/datadog/android/ndk/NdkTests.kt

```diff
@@ -0,0 +1,99 @@
+import java.util.regex.Pattern
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+internal class NdkTests {
+
+    @get:Rule
+    val forge = ForgeRule()
+
+    @get:Rule
+    val temporaryFolder = TemporaryFolder()
+
+    companion object {
+        init {
+            System.loadLibrary("datadog-native-lib")
+            System.loadLibrary("datadog-native-lib-test")
+        }
+    }
+
+    @Test
+    fun ndkUnitTests() {
+        if (runNdkUnitTests() != 0) {
+            throw RuntimeException("NDK Unit tests failed")
+        }
+    }
+
+    @Test
+    fun signalHandlerIntegrationTest() {
+
+        // clear the logs dir
+        val signal = forge.anInt(min = 1, max = 32)
+        // we need to keep this this size because we are using a buffer of [30] size in c++ for
+        // the error.signal attribute
+        val signalName = forge.anAlphabeticalString(size = 20)
+        val signalErrorMessage = forge.anAlphabeticalString()
+        runNdkSignalHandlerIntegrationTest(
+            temporaryFolder.root.absolutePath,
+            signal,
+            signalName,
+            signalErrorMessage
+        )
+
+        // we need to give time to native part to write the file
+        // otherwise we will get into race condition issues
+        Thread.sleep(5000)
+
+        // assert the log file
+        val inputStream = temporaryFolder.root.listFiles()?.first()?.inputStream()
+        inputStream?.use {
+            val jsonString = String(it.readBytes(), Charset.forName("utf-8"))
+            val jsonObject = JsonParser.parseString(jsonString).asJsonObject
+            assertThat(jsonObject.getAsJsonPrimitive("service").asString).isEmpty()
+            assertThat(jsonObject.getAsJsonPrimitive("env").asString).isEmpty()
+            assertThat(jsonObject.getAsJsonPrimitive("status").asString)
+                .isEqualTo("emergency")
+            val dateFormatRegEx =
+                Pattern.compile("[0-9]{4}-[0-9]{2}-[0-9]{2}'T'[0-9]{2}:[0-9]{2}:[0-9]{2}.000Z")
+            assertThat(jsonObject.getAsJsonPrimitive("date").asString)
+                .matches(dateFormatRegEx)
+            assertThat(jsonObject.getAsJsonPrimitive("env").asString).isEmpty()
+            assertThat(jsonObject.getAsJsonPrimitive("message").asString)
+                .isEqualTo("Native crash detected")
+            assertThat(jsonObject.getAsJsonPrimitive("error.message").asString)
+                .isEqualTo(signalErrorMessage)
+            assertThat(jsonObject.getAsJsonPrimitive("error.signal").asString)
+                .isEqualTo("$signalName: $signal")
+            assertThat(jsonObject.getAsJsonPrimitive("error.kind").asString)
+                .isEqualTo("Native")
+            assertThat(jsonObject.getAsJsonPrimitive("logger.name").asString)
+                .isEqualTo("crash")
```

Comment:
Why don't you reuse the existing JsonAssertion utilities ?

Reply:
Good point...I actually forgot we had those ;)